Sources/ShowAttachments.php

```diff
@@ -120,8 +120,11 @@ function showAttachment()
 		// ensure variant attachment compatibility
 		$filePath = pathinfo($file['filePath']);
 
-		$file['filePath'] = !file_exists($file['filePath']) && isset($filePath['extension']) ? substr($file['filePath'], 0, -(strlen($filePath['extension']) + 1)) : $file['filePath'];
-		$file['etag'] = '"' . md5_file($file['filePath']) . '"';
+		$file['exists'] = file_exists($file['filePath']);
+		$file['filePath'] = !$file['exists'] && isset($filePath['extension']) ? substr($file['filePath'], 0, -(strlen($filePath['extension']) + 1)) : $file['filePath'];
+		$file['mtime'] = $file['exists'] ? filemtime($file['filePath']) : 0;
+		$file['size'] = $file['exists'] ? filesize($file['filePath']) : 0;
+		$file['etag'] = '"' . sha1(md5_file($file['filePath']) . '|' . $file['mtime']) . '"';
```

Comment:
There's no need to check the modification time if we are already using `md5_file()` to get a checksum of the file contents. There's also no point using `sha1()` to re-hash the value. If you would prefer to use `sha1_file()` to get the file checksum, rather than `md5_file()`, that'd fine, but I don't think it makes much difference for our purposes here.

Reply:
Done.  I rather us use sha1_file as its 40 characters versus 32 of md5_file.  Anything higher than that isn't worth the cpu cycles for a etag that simply determines modified information on a file.